### ISSUE TITLE
"Not enough data present in the training set."

### ISSUE THREAD

**Issue by Cyzzby** [codename0og/codename-rvc-fork-3#14]
It seems that no matter how my training config is set up, no matter how my 15 minute dataset is pre-processed, I am always met with the following error: 
`Not enough data present in the training set. Perhaps you didn't slice the audio files? ( Preprocessing step )`

The only other errors that are present in my log are client socket failures:
```
[W703 15:21:34.000000000 socket.cpp:755] [c10d] The client socket has failed to connect to [DESKTOP-LQFDIG4]:46117 (system error: 10013 - An attempt was made to access a socket in a way forbidden by its access permissions.).
[W703 15:21:34.000000000 socket.cpp:755] [c10d] The client socket has failed to connect to [keystone.mwbsys.com]:46117 (system error: 10049 - The requested address is not valid in its context.).
```

This is the full log itself: 
```
Starting preprocess with 16 processes...
100%|████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:41<00:00, 41.47s/it]
Preprocess completed in 41.48 seconds on 00:15:18 seconds of audio.
[Config] Running on CUDA, precision loaded from config: bf16
Starting pitch extraction with 16 cores on cuda:0 using rmvpe...
[Config] Running on CUDA, precision loaded from config: bf16
Pitch extraction completed in 4.37 seconds.
Starting embedding extraction with 16 cores on cuda:0...
[Config] Running on CUDA, precision loaded from config: bf16
100%|████████████████████████████████████████████████████████████████████████████████| 342/342 [00:07<00:00, 45.99it/s]
Embedding extraction completed in 11.53 seconds.
Saved index file 'E:\AI\Codename-RVC-Fork\logs\AnakinLucasV2\AnakinLucasV2.index'
    ██████  PRECISION: BrainFloat16 AMP                   ██████
    ██████  cudnn.benchmark: True                         ██████
    ██████  cudnn.deterministic: False                    ██████
    ██████  Optimizer used: AdamW                         ██████
    ██████  Uncertainty loss balancer: No                 ██████
[W703 15:21:34.000000000 socket.cpp:755] [c10d] The client socket has failed to connect to [DESKTOP-LQFDIG4]:46117 (system error: 10013 - An attempt was made to access a socket in a way forbidden by its access permissions.).
[W703 15:21:34.000000000 socket.cpp:755] [c10d] The client socket has failed to connect to [keystone.mwbsys.com]:46117 (system error: 10049 - The requested address is not valid in its context.).
Not enough data present in the training set. Perhaps you didn't slice the audio files? ( Preprocessing step )
```

I am using the default settings, albeit I've changed the sampling rate from 48 to 40, the architecture to Fork/Applio, vocoder to RefineGAN, process effects is unticked, saving frequency is 5 instead of 1, total epochs is 1500, and the TF32 precision is used as well as BF16. I've tried changing the batch size, switching vocoders and precisions, reinstalling the fork, running it as adminstrator, ticking on and off TF32, and combining all of the flacs in my dataset into one, but none of them worked and I am at a complete loss as to what I should do next.

**Comment by codename0og** [comment]
I need you to provide me few more details:

1. Are you doing a mulit-gpu training? (

2. Are you training on your own pc or using some gpu-renting service?

3. Upon launching the fork, you may be prompted at any time to allow Python through the Windows Firewall. Did you allow?

4. After preprocessing, feature extraction and index building, did you confirm you actually have all the needed files in the model's folder?

5. Have you tried on any other dataset? If not, try and let me know.

ps. Always fuse all samples / voice-lines or whatever into one file and then do silence-truncation.
This step is now essential, especially since validation has been added.

**Comment by Cyzzby** [comment]
> I need you to provide me few more details:
> 
> 1. Are you doing a mulit-gpu training? (
> 2. Are you training on your own pc or using some gpu-renting service?
> 3. Upon launching the fork, you may be prompted at any time to allow Python through the Windows Firewall. Did you allow?
> 4. After preprocessing, feature extraction and index building, did you confirm you actually have all the needed files in the model's folder?
> 5. Have you tried on any other dataset? If not, try and let me know.
> 
> ps. Always fuse all samples / voice-lines or whatever into one file and then do silence-truncation. This step is now essential, especially since validation has been added.

1. No. I only have my one GPU; an RTX 3050 to be specific.
2. It's being done locally. 
3. I didn't get the prompt whatsoever on 3.2.0, but I did on 3.1.7-rev1 as it was beginning (and succeeded with starting up) the training process.
4. In extension to #3, I did notice the f0 folders did not have any files in them when training with 3.2.0, whereas 3.1.7-rev1 did. (Sorry, should have included it within my post.)
5. No and honestly, from my experience, it shouldn't as this was the first time I have ever received this error in all of the time I've been training voice models.

Duly noted! I usually get a bit lazy where I could with my datasets, but I will keep that in mind from now on.
